### ISSUE TITLE
How to attach SwiftMessagesSegue viewcontroller to keyboard when keyboard appears.

### ISSUE THREAD

**Issue by stshelton** [SwiftKickMobile/SwiftMessages#275]
I'm having trouble getting the view controller presented by SwiftMessagesSegue to stick to the top of the keyboard. When I present keyboard I tried moving the view up with the keyboardwillshowNotification observer and tried changing bottom constraint to take in keyboard's height neither have worked. What would be the best way to show view controller presented by SwiftMessagesSegue to stick to the top of the keyboard.

**Comment by wtmoose** [comment]
This seems like a good feature to add. Lemme get back to you.

**Comment by n3d1117** [comment]
Are there any updates on this?

**Comment by wtmoose** [comment]
I haven’t worked on this, yet but thanks for the reminder.

**Comment by CodingAleCR** [comment]
I'm just showing a plain old SwiftMessageView as an ActionSheet, the thing is that it has a UITextView on it and the keyboard is shown on top of it. Do you have any suggestion as to how I could move the SwiftMessageView to be on top of the Keyboard?

**Comment by CodingAleCR** [comment]
Nevermind, I just figured I needed to adjust the bottom constraint for my ContentView whenever KeyboardDidHide or KeyboardWillShow notifications are received. Just laying it out there if anyone else need's it. Anyway, thanks.

**Comment by n3d1117** [comment]
Thanks for sharing @CodingAleCR. I'm using a programmatic `SwiftMessagesSegue` with `.centered` layout, presenting a `UINavigationController`, and I can't seem to figure out exactly what constraint should be changed on keyboard show/hide notifications.

**Comment by wtmoose** [comment]
I added first class support for this. If anyone wants toto check the head of `master` and let me know how it works out, I'll wait a bit before releasing.

````swift
// For view controllers
segue.keyboardTrackingView = KeyboardTrackingView()

/// Or, for views
config.keyboardTrackingView = KeyboardTrackingView()
````

`KeyboardTrackingView` its a view that automatically adjusts it's height based on keyboard events and can be used with Auto Layout to make views avoid the keyboard. When you supply an instance of `KeyboardTrackingView` as shown above, SwiftMessages automatically installs it such that the entire message view or view controller avoids the keyboard.

@stshelton @CodingAleCR @n3d1117

**Comment by n3d1117** [comment]
@wtmoose Thank you very much for the update. Your solution is working fine for `.bottom` presentation style, however I'm using a `.centered` style View Controller with a limited height, and the result is the following:
<p align="center">
  <img src="https://user-images.githubusercontent.com/11541888/58056284-eb85c480-7b60-11e9-9d87-6c427114331c.png" height="400">
  <img src="https://user-images.githubusercontent.com/11541888/58056285-eb85c480-7b60-11e9-9584-ef3c565e1d59.png" height="400">
</p>

I think ideally the centered view should adjust its bottom constraint not relative to the whole keyboard height, but to the difference between the keyboard height and the view's bottom spacing. Do you think there's a way this can be achieved? 

Thanks again for your time

**Comment by wtmoose** [comment]
@n3d1117 Yep, I agree 100%. That is what I intended, but I didn't think my implementation through very well. Try the latest update and thanks for helping me test!

**Comment by n3d1117** [comment]
@wtmoose Great stuff! Working perfectly now. Thanks a lot

**Comment by wtmoose** [comment]
Fixed in 7.0.0
